mllib/src/test/scala/org/apache/spark/ml/classification/RandomForestClassifierSuite.scala

```diff
@@ -259,6 +263,37 @@ class RandomForestClassifierSuite extends MLTest with DefaultReadWriteTest {
     })
   }
 
+  test("training with sample weights") {
+    val df = binaryDataset
+    val numClasses = 2
+    // (numTrees, maxDepth, subsamplingRate, fractionInTol)
+    val testParams = Seq(
+      (20, 5, 1.0, 0.96),
+      (20, 10, 1.0, 0.96),
+      (20, 10, 0.95, 0.96)
+    )
```

Comment:
I guess maybe also add different impurity in testParams?

Reply:
Maybe also test a special case numTrees = 1?

Reply:
with numTrees==1, RF is exactly the DecisionTree, which is already tested in DecisionTreeClassifierSuite/DecisionTreeRegressorSuite.

> I guess maybe also add different impurity in testParams?

I guess current tests maybe enough, Testsuites for DT/GBT do not test impurity.

Reply:
The reason I suggested testing different impurities is because when calculating best split, the impurity path (both ```entropy``` and ```gini```) is affected by sample weight. However, after taking a look at the DecisionTree test, I saw both ```entropy``` and ```gini``` are tested with sample weight there, so this is already covered in DecisionTree test, no need to test here.